plugin/trino-openlineage/src/main/java/io/trino/plugin/openlineage/OpenLineageListenerModule.java

```diff
@@ -0,0 +1,110 @@
+import io.openlineage.client.transports.HttpTransport;
+import io.openlineage.client.transports.TokenProvider;
+import io.trino.plugin.openlineage.config.OpenLineageClientHttpTransportConfig;
+import io.trino.plugin.openlineage.config.OpenLineageListenerConfig;
+
+import java.net.URI;
+import java.util.Optional;
+
+import static com.google.inject.multibindings.OptionalBinder.newOptionalBinder;
+import static io.airlift.configuration.ConditionalModule.conditionalModule;
+import static io.airlift.configuration.ConfigBinder.configBinder;
+import static java.lang.String.format;
+
+public class OpenLineageListenerModule
+        extends AbstractConfigurationAwareModule
+{
+    private static class ApiKeyTokenProvider
+            implements TokenProvider
+    {
+        private final String token;
+
+        public ApiKeyTokenProvider(String token)
+        {
+            this.token = token;
+        }
+
+        @Override
+        public String getToken()
+        {
+            return format("Bearer %s", this.token);
+        }
+    }
+
+    @Override
+    protected void setup(Binder binder)
+    {
+        binder.bind(OpenLineageListener.class).in(Scopes.SINGLETON);
+
+        configBinder(binder).bindConfig(OpenLineageListenerConfig.class);
+
+        newOptionalBinder(binder, OpenLineageClientHttpTransportConfig.class);
+
+        install(conditionalModule(
+                OpenLineageListenerConfig.class,
+                config -> config.getTransport()
+                        .equals(OpenLineageTransport.HTTP),
+                conditionalBinder -> configBinder(conditionalBinder).bindConfig(OpenLineageClientHttpTransportConfig.class)));
+    }
+
+    @Provides
+    @Singleton
+    private OpenLineageClient getClient(OpenLineageListenerConfig listenerConfig, Optional<OpenLineageClientHttpTransportConfig> httpTransportConfig)
+            throws Exception
+    {
+        OpenLineageClient.Builder clientBuilder = OpenLineageClient.builder();
+
+        if (listenerConfig.getTransport().equals(OpenLineageTransport.HTTP) && httpTransportConfig.isPresent()) {
+            OpenLineageClientHttpTransportConfig config = httpTransportConfig.get();
+            clientBuilder.transport(
+                    new HttpTransport(
+                            new HttpConfig(
+                                    new URI(config.getUrl()),
+                                    config.getEndpoint(),
+                                    null,
+                                    (int) config.getTimeout().toMillis(),
+                                    config.getApiKey().isPresent() ? new OpenLineageListenerModule.ApiKeyTokenProvider(config.getApiKey().get()) : null,
+                                    config.getUrlParams(),
+                                    config.getHeaders())));
+        }
+        else if (listenerConfig.getTransport() == OpenLineageTransport.CONSOLE) {
+            clientBuilder.transport(new ConsoleTransport());
```

Comment:
It is possible for us to introduce some sort of a interface like `MongoClientSettingConfigurator` we have in MongoDB which could use 
```
public interface OpenLineageClientConfigurator
{
    void configure( OpenLineageClient.Builder builder);
}
```

The logic related to each transport layer could be specific to each module.

Reply:
I think as we have just one configurable transport for now I would leave it as is - with introducing subsequent ones this of course can be generalized.

Reply:
It is technically two right `CONSOLE` and `HTTP` as far as new one - I guess Kafka - the implementation and testing could be a bit tricky and I think handling this prep work as a part of that PR would make it a bit complex. Feel free to ping me in slack if you need any help on abstracting it out.

Reply:
ok
- added `OpenLineageTransportBuilder` interface
- implemented above for console and http
- added conditional module installation in `OpenLineageListenerModule`

I think it's now quite generic and clear how to extend with new transport types